File: javer/src/main/java/com/uuhnaut69/javer/model/BaseEntity.java
package com.uuhnaut69.javer.model;

import lombok.Getter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.sql.Timestamp;
import java.util.UUID;

@Getter
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public class BaseEntity {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private UUID id;

  @CreatedDate
  @Column(nullable = false)
  private Timestamp createdDate;
}

File: api/kover/src/main/kotlin/com/uuhnaut69/kover/model/BaseEntity.kt
package com.uuhnaut69.kover.model

import lombok.Getter
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.jpa.domain.support.AuditingEntityListener
import java.sql.Timestamp
import java.util.*
import javax.persistence.*

@Getter
@MappedSuperclass
@EntityListeners(AuditingEntityListener::class)
open class BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: UUID? = null

    @CreatedDate
    @Column(nullable = false)
    var createdDate: Timestamp? = null

}